riptide-spring-boot-autoconfigure/src/test/java/org/zalando/riptide/autoconfigure/OpenTelemetryPluginFactoryTest.java

```diff
@@ -22,62 +22,109 @@ public class OpenTelemetryPluginFactoryTest {
 
     private static final Field SPAN_DECORATOR_FIELD = ReflectionUtils.findField(OpenTelemetryPlugin.class, "spanDecorator");
     private static final Field DECORATORS_FIELD = ReflectionUtils.findField(CompositeSpanDecorator.class, "decorators");
+    private static final Field PROPAGATE_FLOW_ID_FIELD = ReflectionUtils.findField(FlowIdSpanDecorator.class, "propagateFlowId");
 
     static {
         SPAN_DECORATOR_FIELD.setAccessible(true);
         DECORATORS_FIELD.setAccessible(true);
+        PROPAGATE_FLOW_ID_FIELD.setAccessible(true);
     }
 
     @ParameterizedTest
     @CsvSource({
         "true, RetrySpanDecorator should be added when retry is enabled",
         "false, RetrySpanDecorator should not be added when retry is disabled"
     })
-    void shouldCreatePluginWithRetrySpanDecoratorWhenClientRetryEnabled(
-            final boolean isRetryEnabled,
-            final String message
+    void shouldAddRetrySpanDecoratorWhenEnabled(
+        final boolean isRetryEnabled,
+        final String message
     ) throws IllegalAccessException {
-        final Plugin plugin = OpenTelemetryPluginFactory.create(createTestClient(isRetryEnabled));
+        final Plugin plugin = OpenTelemetryPluginFactory.create(createTestClient(isRetryEnabled, false));
 
         assertNotNull(plugin);
         assertInstanceOf(OpenTelemetryPlugin.class, plugin);
 
-        final Optional<SpanDecorator> innerCompositeDecorator = StreamSupport.stream(
-                ((Iterable<SpanDecorator>) DECORATORS_FIELD.get(SPAN_DECORATOR_FIELD.get(plugin))).spliterator(), false
-        ).filter(decorator -> decorator instanceof CompositeSpanDecorator).findFirst();
+        final Optional<CompositeSpanDecorator> innerCompositeDecorator = getDecorator(
+            (SpanDecorator) SPAN_DECORATOR_FIELD.get(plugin),
+            CompositeSpanDecorator.class
+        );
 
         assertThat(innerCompositeDecorator).isPresent();
-        assertThat(
-            StreamSupport.stream(
-                ((Iterable<SpanDecorator>) DECORATORS_FIELD.get(innerCompositeDecorator.get())).spliterator(), false
-            ).anyMatch(decorator -> decorator instanceof RetrySpanDecorator)
-        ).withFailMessage(message).isEqualTo(isRetryEnabled);
+        assertEquals(
```

Comment:
I know that this test class already used JUnit assertions, but ideally it should only use one assertion library, so I would ask you to switch to assertJ assertions for the changes you add (no need to replace the existing assertions, allow would be nice to have).